### PR TITLE
`explicit-length-check`: Ignore known, non-number length properties

### LINKED ISSUE #1262: [explicit-length-check]: misunderstanding a string an index
eslint-plugin-unicorn is fantastic! Keep up the good work.

image
In this screenshot you can see property 'size' is a string. I think it's safe as even `"0"` resolves to truthy.

explicit-length-check

```js
const blah = { size: 'small' }
blah.size && true
```

### PR DESCRIPTION
**Pull request by bmish**
Fixes #1262.

### CODE REVIEW

**Review by fisker — CHANGES_REQUESTED** [review]
Looks good, but seems you forgot to update snapshot.

File: rules/explicit-length-check.js [hunk 1]
```diff
@@ -154,6 +154,12 @@ function create(context) {
 			let node;
 			let autoFix = true;
 
+			const staticValue = getStaticValue(lengthNode, context.getScope());
+			if (staticValue && typeof staticValue.value !== 'number') {
+				// Ignore known, non-number length properties.
```
Comment (fisker) [inline rules/explicit-length-check.js:159]: Let's get this even stricter, only allow positive integers.

Comment (bmish) [reply rules/explicit-length-check.js]: Good point! Fixed.

Comment (fisker) [reply rules/explicit-length-check.js]: Actually, I mean use `Number.isInteger()` exclude `1.2` too.

Comment (bmish) [reply rules/explicit-length-check.js]: Fixed.

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
@bmish  Can you add `NaN` and `Infinity` to the test?

**Pull request merged**